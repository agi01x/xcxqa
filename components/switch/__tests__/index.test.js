import React from 'react';
import Switch from '..';
import { mount } from 'enzyme';
import focusTest from '../../../tests/shared/focusTest';

describe('Switch', () => {
  focusTest(Switch);

  it('should has click wave effect', async () => {
    const wrapper = mount(<Switch />);
    wrapper.find('.ant-switch').getDOMNode().click();
    await new Promise(resolve => setTimeout(resolve, 0));
    expect(wrapper.render()).toMatchSnapshot();
  });
});
