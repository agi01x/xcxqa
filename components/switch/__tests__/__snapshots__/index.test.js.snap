// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should has click wave effect 1`] = `
<button
  ant-click-animating="true"
  aria-checked="false"
  class="ant-switch"
  role="switch"
  type="button"
>
  <span
    class="ant-switch-inner"
  />
  <div
    class="ant-click-animating-node"
    style="border-color: buttonface;"
  />
</button>
`;
